source/grid/grid_in.cc

```diff
@@ -2542,6 +2566,162 @@ void GridIn<dim, spacedim>::read_tecplot(std::istream &)
 }
 
 
+
+template <int dim, int spacedim>
+void GridIn<dim, spacedim>::read_assimp(const std::string &filename,
+                                        const unsigned int mesh_index,
+                                        const bool remove_duplicates,
+                                        const double tol,
+                                        const bool ignore_unsupported_types)
+{
+#ifdef DEAL_II_WITH_ASSIMP
+  // Only good for surface grids.
+  AssertThrow(dim<3, ExcImpossibleInDim(dim));
+
+  // Create an istance of the Importer class
+  Assimp::Importer importer;
+
+  // And have it read the given file with some  postprocessing
+  const aiScene *scene = importer.ReadFile( filename.c_str(),
+                                            aiProcess_RemoveComponent   |
+                                            aiProcess_JoinIdenticalVertices   |
+                                            aiProcess_ImproveCacheLocality  |
+                                            aiProcess_SortByPType   |
+                                            aiProcess_OptimizeGraph   |
+                                            aiProcess_OptimizeMeshes);
+
+  // If the import failed, report it
+  AssertThrow(scene, ExcMessage(importer.GetErrorString()))
+
+  AssertThrow(scene->mNumMeshes, ExcMessage("Input file contains no meshes."));
+
+  AssertThrow((mesh_index == numbers::invalid_unsigned_int) ||
+              (mesh_index < scene->mNumMeshes),
+              ExcMessage("Too few meshes in the file."));
+
+  int start_mesh = (mesh_index == numbers::invalid_unsigned_int ?
+                    0 : mesh_index);
+  int end_mesh = (mesh_index == numbers::invalid_unsigned_int ?
+                  scene->mNumMeshes : mesh_index+1);
+
+  // Deal.II objects are created empty, and then filled with imported file.
+  std::vector<Point<spacedim> > vertices;
+  std::vector<CellData<dim> > cells;
+  SubCellData subcelldata;
+
+  // A series of counters to merge cells.
+  unsigned int v_offset=0;
+  unsigned int c_offset=0;
+
+  // The index of the mesh will be used as a material index.
+  for (int m=start_mesh; m<end_mesh; ++m)
+    {
+      const aiMesh *mesh = scene->mMeshes[m];
+
+      // Check that we know what to do with this mesh, otherwise just
+      // ignore it
+      if ( (dim == 2) && mesh->mPrimitiveTypes != aiPrimitiveType_POLYGON)
+        {
+          std::stringstream s;
+          s << "Incompatible mesh "  << m
+            << "/" << scene->mNumMeshes;
+          AssertThrow(ignore_unsupported_types, ExcMessage(s.str()));
+          continue;
```

Comment:
You could compress all of that into
```
  AssertThrow (..., ExcMessage ("Incompatible mesh " + Utilities::int_to_string(m) + "/' + Utilities::int_to_string(scene->mNumMeshes));
```

Reply:
I don't like using too much of `int_to_string` stuff... if you really want me to, I will make that change... :)

Reply:
It's mostly about the issue that you build a string that you don't actually use in almost all cases.